item-impl/src/main/java/com/example/auction/item/impl/ItemRepository.java

```diff
@@ -34,7 +34,6 @@
     @Inject
     public ItemRepository(CassandraSession session, ReadSide readSide) {
         this.session = session;
-        session.underlying().thenAccept(s -> registerCodec(s, new EnumNameCodec<>(ItemStatus.class)));
```

Comment:
I wasn't really sure if the constructor was the best place to do this, but I'm curious about your reason for moving it.